File: nodejs/src/controllers/demoController.js
const { getUsers } = require('../models/user');

const hello = (req, res) => {
    res.send('Hello World (Nodejs / Express)');
};

const users = async (req, res) => {
    try {
        const users = await getUsers();
        res.json(users);
    } catch (error) {
        console.error(error);
        res.status(500).json({ error: 'Database error' });
    }
};

const sleep = (req, res) => {
    const seconds = parseInt(req.params.seconds, 10);
    setTimeout(() => {
        res.send(`sleep ${seconds}s`);
    }, seconds * 1000);
};

const randomStatus = (req, res) => {
    const statuses = [
        { code: 200, message: 'OK' },
        { code: 201, message: 'Created' },
        { code: 202, message: 'Accepted' },
        { code: 204, message: 'No Content' },
        { code: 400, message: 'Bad Request' },
        { code: 401, message: 'Unauthorized' },
        { code: 403, message: 'Forbidden' },
        { code: 404, message: 'Not Found' },
        { code: 500, message: 'Internal Server Error' },
        { code: 501, message: 'Not Implemented' },
        { code: 502, message: 'Bad Gateway' },
        { code: 503, message: 'Service Unavailable' },
    ];
    const randomStatus = statuses[Math.floor(Math.random() * statuses.length)];
    res.status(randomStatus.code).send(randomStatus.message);
};

const exception = (req, res) => {
    const currentTime = new Date().toISOString();
    console.error(`exception called at ${currentTime}`);
    res.status(500).send(`exception called at ${currentTime}`);
};

module.exports = { hello, users, sleep, randomStatus, exception };

File: java/src/main/java/com/example/demo/controller/DemoController.java
package com.example.demo.controller;

import com.example.demo.model.User;
import com.example.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Random;
import java.util.logging.Logger;

@RestController
public class DemoController {

    private static final Logger logger = Logger.getLogger(DemoController.class.getName());

    @Autowired
    private UserRepository userRepository;

    @GetMapping("/")
    public String hello() {
        return "Hello World (Java / Springboot)";
    }

    @GetMapping("/sleep/{seconds}")
    public String sleep(@PathVariable int seconds) throws InterruptedException {
        Thread.sleep(seconds * 1000);
        return "sleep " + seconds + "s";
    }

    @GetMapping("/status/random")
    public ResponseEntity<String> randomStatus() {
        int[] statusCodes = {200, 201, 202, 204, 400, 401, 403, 404, 500, 501, 502, 503};
        String[] statusMessages = {"OK", "Created", "Accepted", "No Content", "Bad Request", "Unauthorized", "Forbidden", "Not Found", "Internal Server Error", "Not Implemented", "Bad Gateway", "Service Unavailable"};

        Random random = new Random();
        int index = random.nextInt(statusCodes.length);

        int statusCode = statusCodes[index];
        String statusMessage = statusMessages[index];

        return ResponseEntity.status(statusCode).body(statusMessage);
    }

    @GetMapping("/exception")
    public String exception() {
        String errMessage = "Internal Server Error - Manual Exception";
        logger.severe(errMessage);
        throw new RuntimeException(errMessage);
    }
}
